pyomo/repn/plugins/gams_writer.py

```diff
@@ -143,56 +231,79 @@ def _print_model( self, model, output_file, solver_capability, var_list,
         for con in model.component_data_objects(Constraint, active=True):
             if con.body.is_fixed():
                 continue
+
+            # Ensure GAMS will not encounter domain violations in presolver
+            # operations at current values, which are None (0) by default
+            # Used to handle log and log10 violations, for example
+            try:
+                value(con.body)
+            except:
+                raise RuntimeError("While evaluating: %s\n\tProblemWriter_gams"
+                                   " encountered an error. Ensure initial"
+                                   " variable values\n\tdo not violate"
+                                   " any domains (are you using log/log10?)"
+                                 % con.name)
+
             if linear:
                 if con.body.polynomial_degree() not in linear_degree:
                     linear = False
-            body = StringIO()
-            con.body.to_string(body)
-            cName=symbolMap.getSymbol(con, labeler)
+            body = str(con.body)
+            cName = symbolMap.getSymbol(con, con_labeler)
             if con.equality:
                 constraint_names.append('%s' % cName)
                 ConstraintIO.write('%s.. %s =e= %s;\n' % (
-                    constraint_names[-1], 
-                    body.getvalue(),
-                    value(con.upper),
+                    constraint_names[-1],
+                    body,
+                    value(con.upper)
                 ))
             else:
                 if con.lower is not None:
                     constraint_names.append('%s_lo' % cName)
                     ConstraintIO.write('%s.. %s =l= %s;\n' % (
-                        constraint_names[-1], 
+                        constraint_names[-1],
                         value(con.lower),
-                        body.getvalue(),
+                        body
                     ))
                 if con.upper is not None:
                     constraint_names.append('%s_hi' % cName)
                     ConstraintIO.write('%s.. %s =l= %s;\n' % (
-                        constraint_names[-1], 
-                        body.getvalue(),
-                        value(con.upper),
+                        constraint_names[-1],
+                        body,
+                        value(con.upper)
                     ))
 
-        obj = list( model.component_data_objects(Objective, active=True) )
+        obj = list(model.component_data_objects(Objective, active=True))
         if len(obj) != 1:
             raise RuntimeError(
                 "GAMS writer requires exactly one active objective (found %s)"
-                % (len(obj),))
+                % (len(obj)))
         obj = obj[0]
+
+        # Same domain violation check as above
```

Comment:
The 3 places the domain check occur are for the constraints, the objective, and Expression objects. They are written in 3 different places because those are encountered and prepared for the output file in different places, so I needed to write checks for those groups individually. Perhaps I'll make a helper function since they're all the same.